### PR TITLE
Rename function: matlab.palette() → palette_matlab()

### LINKED ISSUE #299: Rename functions (1): palette_
This issue complements https://github.com/cbeleites/hyperSpec/issues/208#issuecomment-684085699.

In the development version of **`hyperSpec`**, I found 4 functions/objects that define color palettes.
I suggest using consistent naming (prefix **`palette_`**) for all palette-related objects:

Status | Current name          | Suggested name
:---:|----------------------|-------------------
Done in  #313  | ~~`CB7`~~                 |  ~~`palette_colorblind`~~
✔️ | `matlab.dark.palette` | `palette_matlab_dark`
✔️ | `matlab.palette`      | `palette_matlab`
✔️ | `alois.palette`       | `palette_alois`

### PR DESCRIPTION
**Pull request by sangttruong**
In the development version of hyperSpec, I found 3 functions/objects that define color palettes.
Based on the suggestions, I rename all palette objects (prefix palette_)

Resolve #299

### CODE REVIEW

**Review by GegznaV — CHANGES_REQUESTED** [review]
1) If user-visible changes are introduced, they must be properly described in the `NEWS.md` file. Did you read `CONTRIBUTION.md`?
2) Please avoid committing files that are not related to the issue. I mean `hyperSpec_0.100.0.zip` this time. Please, remove this file and do not include it in the future.
3) Some tests do not pass on the old release of R (v.3.6). I'm not sure what is the reason. 
4) There are 3 renamed functions, so it would have been better if there were 3 separate commits (you will not fix this now, but adhere to this guideline in the future).
5) The short commit messages should be self-explanatory, Not, e.g., "Resolve #299", but e.g., "Rename function: matlab.palette() → palette_matlab()" (you will not fix it now, but keep that in mind for the future).

**Review by bryanhanson — COMMENTED** [review]
Regarding the unit tests in `palette_matlab.R` my own opinion is that testing this sort of thing is unnecessary.  Either the palettes exist, or they don't.  The tests in this PR verify that one entry in the palette is the expected one, but I can't imagine a situation where this would change, so I'm not really sure what we are trying to defend against.  If it were up to me I'd simply remove the unit tests for this function.

File: hyperSpec/R/palette_matlab.R [hunk 1]
```diff
@@ -82,13 +82,43 @@ matlab.dark.palette <- function(n = 100, ...) {
...
+#' plotmap(faux_cell[, , 1200], col.regions = palette_alois())
 #'
-alois.palette <- function(n = 100, ...) {
+palette_alois <- function(n = 100, ...) {
   colorRampPalette(c("black", "blue", "cyan", "green", "yellow", "red"), ...)(n)
 }
+
+# Unit tests -----------------------------------------------------------------
+#' @import hySpc.testthat
+library(hySpc.testthat)
```
Comment (GegznaV) [inline hyperSpec/R/palette_matlab.R:98]: Function `library()` must not be used in packages.

File: hyperSpec/R/palette_matlab.R [hunk 2]
```diff
@@ -82,13 +82,43 @@ matlab.dark.palette <- function(n = 100, ...) {
...
+hySpc.testthat::test(palette_matlab_dark) <- function() {
+  context("palette_matlab_dark")
+  test_that("test palette matlab dark color", {
```
Comment (GegznaV) [inline hyperSpec/R/palette_matlab.R:101]: The whole line should be as a fluent and logical sentence, e.g., 
"Test that *palette_matlab_dark() generates correct colors*"
"Test that *palette_matlab_dark() generates works*"
etc.
This sentence should describee the unit of functionality that is tested by the unit test.

File: hyperSpec/R/palette_matlab.R [hunk 3]
```diff
@@ -82,13 +82,43 @@ matlab.dark.palette <- function(n = 100, ...) {
...
+  context("palette_matlab")
+  test_that("test palette matlab", {
+    pal <- palette_matlab()
+    expect_true(pal[1] == "#0000FF" | pal[1] == "#0000FFFF")
+    expect_true(pal[2] == "#000AFF" | pal[2] == "#000AFFFF")})
+}
+
+hySpc.testthat::test(palette_alois) <- function() {
+  context("palette_alois")
+  test_that("test palette alois", {
```
Comment (GegznaV) [inline hyperSpec/R/palette_matlab.R:117]: The whole line should be as a fluent and logical sentence.

File: hyperSpec/R/palette_matlab.R [hunk 4]
```diff
@@ -82,13 +82,43 @@ matlab.dark.palette <- function(n = 100, ...) {
...
+  context("palette_matlab_dark")
+  test_that("test palette matlab dark color", {
+    dark <- palette_matlab_dark()
+    expect_equal(dark[1], "#0000FF")
+    expect_equal(dark[2], "#0005FF")})
+}
+
+hySpc.testthat::test(palette_matlab) <- function() {
+  context("palette_matlab")
+  test_that("test palette matlab", {
```
Comment (GegznaV) [inline hyperSpec/R/palette_matlab.R:109]: The whole line should be as a fluent and logical sentence

File: hyperSpec/R/palette_matlab.R [hunk 5]
```diff
@@ -82,13 +82,43 @@ matlab.dark.palette <- function(n = 100, ...) {
...
+library(hySpc.testthat)
+hySpc.testthat::test(palette_matlab_dark) <- function() {
+  context("palette_matlab_dark")
+  test_that("test palette matlab dark color", {
+    dark <- palette_matlab_dark()
+    expect_equal(dark[1], "#0000FF")
+    expect_equal(dark[2], "#0005FF")})
+}
+
+hySpc.testthat::test(palette_matlab) <- function() {
```
Comment (GegznaV) [inline hyperSpec/R/palette_matlab.R:107]: Tests should go in the same order as functions (matlab, matlab dark, alois)

**Review by GegznaV — CHANGES_REQUESTED** [review]
There is some general feedback on unit tests.

**Comment by GegznaV** [comment]
@bryanhanson:
> Either the palettes exist, or they don't. 

Some palettes depand on other functions of "R". And, e.g., in version 4.0 "R" changed some default colors.

File: hyperSpec/R/palette_matlab.R [hunk 6]
```diff
@@ -82,13 +82,39 @@ matlab.dark.palette <- function(n = 100, ...) {
...
+    expect_equal(dark[2], "#0005FF")})
+}
+
+hySpc.testthat::test(palette_alois) <- function() {
+  context("palette_alois")
+  test_that("palette_alois() works", {
+    alois <- palette_alois()
+    expect_equal(alois[1], "#000000")
+    expect_equal(alois[2], "#00000C")})
+}
```
Comment (GegznaV) [inline hyperSpec/R/palette_matlab.R:120]: Please, leave a blank line at the end of each file. In some circumstances, the absence of a new line may result in unwanted issues.

image

**Review by GegznaV — CHANGES_REQUESTED** [review]
Please, add a new line at the end of files, where needed.

**Review by bryanhanson — APPROVED** [review]

**Pull request merged**